### PR TITLE
feat(Core/Modules): Preparation to implement progression-system module.

### PR DESCRIPTION
**Pull request by UltraNix**
## Changes Proposed:
- Added databse scripts to Script manager
- Added new hook: `OnAfterDatabasesLoaded`
- Allows to load more than one script object per script name
- Merge #8825

## How to Test the Changes:

Nothing to test.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
2 failures:

> src/server/scripts/CMakeLists.txt:356:    
Remove whitespace at the end of the lines above

> PlayerCommand.h:21:23: fatal error: unknown type name 'ChatHandler'
    static bool Learn(ChatHandler* handler, Player* targetPlayer, uint32 spell, char const* all);

meanwhile @Winfidonarleyan and @Yehonal please review this PR

**Comment by UltraNix** [comment]
I don't know why build fails. 
```PlayerCommand.h:21:23: fatal error: unknown type name 'ChatHandler'
static bool Learn(ChatHandler* handler, Player* targetPlayer, uint32 spell, char const* all);```
It's not related to thisPR at all.

```src/server/scripts/CMakeLists.txt:356:
Remove whitespace at the end of the lines above```
I don;t see any whitespace there.

File: src/server/database/Updater/UpdateFetcher.cpp [hunk 1]
```diff
@@ -347,37 +370,51 @@ UpdateResult UpdateFetcher::Update(bool const redundancyChecks,
...
+        if (!toCleanup.empty())
         {
-            LOG_ERROR("sql.updates", "Cleanup is disabled! There were  " SZFMTD " dirty files applied to your database, " \
-                      "but they are now missing in your source directory!", applied.size());
+            if (doCleanup)
+                CleanUp(toCleanup);
+            else
+            {
+                LOG_ERROR("sql.updates",
+                          "Cleanup is disabled! There were  " SZFMTD " dirty files applied to your database, "
```
Comment (Winfidonarleyan) [inline src/server/database/Updater/UpdateFetcher.cpp:414]:
```suggestion
                          "Cleanup is disabled! There were " SZFMTD " dirty files applied to your database, "
```

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by Winfidonarleyan** [comment]
Looks good at first glance

**Comment by Nyeriah** [comment]
@Winfidonarleyan do you have any clues about why would the module build be failing? It's complaining about a file not touched by this PR, I don't understand it. We really need this PR merged as soon as possible :)

**Comment by Winfidonarleyan** [comment]
- [ ] Wait merge #8947

**Comment by Nyeriah** [comment]
Ultranix, please try now

File: src/server/game/Scripting/ScriptMgr.cpp [hunk 2]
```diff
@@ -3148,6 +3150,11 @@ void ScriptMgr::OnHandleDevCommand(Player* player, std::string& argstr)
     FOREACH_SCRIPT(CommandSC)->OnHandleDevCommand(player, argstr);
 }
 
+void ScriptMgr::AfterDatabasesLoaded(uint32 updateFlags)
```
Comment (Yehonal) [inline src/server/game/Scripting/ScriptMgr.cpp:3153]: Correct naming is: ```On<When><Action>``` 

Hence: OnAfterDatabasesLoaded

**Review by Yehonal — CHANGES_REQUESTED** [review]

**Comment by FrancescoBorzi** [comment]
I did some research [here](https://github.com/azerothcore/azerothcore-wotlk/pull/8992) and I found out that it's the [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter) that is causing the issue

cc @UltraNix @sanctum32 @r-o-b-o-t-o  @Nyeriah

**Comment by UltraNix** [comment]
Resolved by https://github.com/azerothcore/mod-chat-transmitter/pull/2/

**Pull request merged**